v2/fb-xcom-api-validator/tests/capabilities.test.ts

```diff
@@ -1,77 +1,85 @@
+import { randomUUID } from 'crypto';
 import Client from '../src/client';
-import { Capabilities } from '../src/client/generated';
+import { ApiError, AssetDefinition, Capabilities, NotFoundError } from '../src/client/generated';
 
-const optionalCapabilitiesComponents = [
-  'historicBalances',
-  'transfers',
-  'transfersBlockchain',
-  'transfersFiat',
-  'transfersPeerAccounts',
-  'trading',
-  'liquidity',
-  'subscriptions',
-];
-
-const requiredCapabilitiesComponents = ['accounts', 'balances'];
+const KNWON_API_VERSIONS = ['0.0.1'];
 
 describe('Capabilities', () => {
+  let client: Client;
+
+  beforeAll(() => {
+    client = new Client();
+  });
+
   describe('/capabilities', () => {
-    let result: Capabilities;
+    const capabilitiesConfig: Capabilities = global.capabilities;
+    let capabilitiesResponse: Capabilities;
 
     beforeAll(async () => {
-      const client = new Client();
-      result = (await client.capabilities.getCapabilities({})) as Capabilities;
+      capabilitiesResponse = (await client.capabilities.getCapabilities({})) as Capabilities;
+    });
+
+    it('should match config capabilities', () => {
+      expect(capabilitiesResponse).toEqual(capabilitiesConfig);
     });
 
     it('should include a valid api version', () => {
-      expect(isValidApiVersion(result.version)).toBe(true);
+      expect(isValidApiVersion(capabilitiesConfig.version)).toBe(true);
     });
+  });
 
-    describe('Component', () => {
-      describe.each(requiredCapabilitiesComponents)('%s (required)', (componentName) => {
-        it('should be listed in response', () => {
-          expect(result.components[componentName]).toBeDefined();
-        });
+  describe('/capabilities/assets', () => {
+    let result: { assets: AssetDefinition[] };
 
-        it('should have a valid value', () => {
-          expect(isValidComponentValue(result.components[componentName])).toBe(true);
-        });
-      });
+    beforeAll(async () => {
+      result = (await client.capabilities.getAdditionalAssets({})) as { assets: AssetDefinition[] };
```

Comment:
Is the cast really needed here?

Reply:
It's due to every request returning a union with GeneralError, which messes up the typing